baseservice/base_service.py

```diff
@@ -0,0 +1,122 @@
+import logging
+import signal
+import threading
+from abc import ABCMeta, abstractmethod
+from enum import Enum, unique
+from typing import Optional
+
+from baseservice.utils import Event
+
+
+@unique
+class ServiceState(Enum):
```

Comment:
Shouldnt we also have a "started" state?

Reply:
well... yes. but if you read the code carefully, there's no place you can fire the 'Started' event, since run_service might be blocking, so you never know when it's started...